### ISSUE TITLE
RangeError: Maximum call stack size exceeded

### ISSUE THREAD

**Issue by mugi-luffy** [nomiddlename/log4js-node#419]
With the new version "1.0.1" my server application crashes at start. The same application worked perfectly with version "0.6.38".
I'm using Node.js "6.9.1".
Here is the error message I get:
```
server/node_modules/log4js/lib/date_format.js:65
    .replace(/y{1,4}/g, vYear)
     ^

RangeError: Maximum call stack size exceeded
    at RegExp.[Symbol.replace] (native)
    at RegExp.[Symbol.replace] (native)
    at String.replace (native)
    at Object.exports.asString (server/node_modules/log4js/lib/date_format.js:65:6)
    at timestampLevelAndCategory (server/node_modules/log4js/lib/layouts.js:86:20)
    at colouredLayout (server/node_modules/log4js/lib/layouts.js:117:10)
    at server/node_modules/log4js/lib/appenders/console.js:8:16
    at Logger.<anonymous> (server/node_modules/log4js/lib/appenders/logLevelFilter.js:11:7)
    at emitOne (events.js:101:20)
    at Logger.emit (events.js:188:7)
```

**Comment by nomiddlename** [comment]
This looks to be the same as #210 and #196 - which looks to be caused by using `replaceConsole: true`. Can you confirm that you're using that in your config?

**Comment by nomiddlename** [comment]
I've tried a few different things to reproduce this error, but I've had no success. Could you provide your config, and ideally some sample code that triggers the error, please?

**Comment by mugi-luffy** [comment]
Thank you for the tip. 
I deleted replaceConsole() from my code and now it seems to be working! I don't know if it's importand but I'am running the server on Ubuntu-Linux.
Here is my code:
```
var log4js = require('log4js');
//log4js.replaceConsole();
this.getLogger = function(name) {
   log4js.configure('./log.json', { reloadSecs: 300 });
   this.logger = log4js.getLogger(name);
   this.logger.debug("Logger successfully initialized.");
   return this.logger;
};
```
and log.json
```
{
  "appenders": [
    {
      "type": "logLevelFilter",
      "level": "INFO",
      "appender": {
        "type": "console"
      }
    },
    {
      "type": "dateFile",
      "filename": "log.txt",
      "pattern": "-yyyy-MM-dd",
      "alwaysIncludePattern": false
    }
  ],
  "level": "ALL"
}
```
In the README.md it says we have to use "replaceConsole: true". Is this no longer the case?

**Comment by nomiddlename** [comment]
You only need to use `replaceConsole` if your code uses `console.log` and you want the output to go to the same place, in the same format, as your log4js logger output. If you're not using console.log, then it's best to just leave it.

You can get rid of the error, and still keep the console replacement by adding `"replaceConsole": true` to your `log.json` file. You shouldn't mix programmatic configuration of `log4js` with loading the config from a file - it's one or the other, not both. Also, it looks like you're calling `log4js.configure` every time you want a logger, is this right? (Hard to tell from that snippet). `configure` should only be called once in an application.

**Comment by mugi-luffy** [comment]
You're absolutely right thanks. It was called every time a logger is returned which is not what I intended. I changed it to this:
myLogger.js
```
function init() {
    try {
        var log4js = require('log4js');
        log4js.configure('./log.json', { reloadSecs: 300 });
        console.log("Logger successfully initialized.")
    }
    catch(ex) {
        console.error("ERROR: logger not initialized: " + ex);
    }

    function getLogger(name) {
            var logger = log4js.getLogger(name);
            logger.info(name, " logger successfully initialized.");
            return logger;
    };

    return {
        getLogger: function(name) { return getLogger(name); }
    };
}

module.exports  = init();
```
Thanks again for the help and for the great work.

**Comment by nomiddlename** [comment]
Fix pushed to npm in version 1.1.0
